<!DOCTYPE html>
<html>
<head>
  <title>Interactive and exploratory visualization of epigenome-wide data</title>
  <meta charset="utf-8">
  <meta name="description" content="Interactive and exploratory visualization of epigenome-wide data">
  <meta name="author" content="Hector Corrada Bravo">
  <meta name="generator" content="slidify" />
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta http-equiv="X-UA-Compatible" content="chrome=1">
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/default.css" media="all" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/phone.css" 
    media="only screen and (max-device-width: 480px)" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/slidify.css" >
  <link rel="stylesheet" href="libraries/highlighters/highlight.js/css/tomorrow.css" />
  <base target="_blank"> <!-- This amazingness opens all links in a new tab. -->  <link rel=stylesheet href="libraries/widgets/bootstrap/css/bootstrap.css"></link>
<link rel=stylesheet href="./assets/css/custom.css"></link>
<link rel=stylesheet href="./assets/css/ribbons.css"></link>

  
  <!-- Grab CDN jQuery, fall back to local if offline -->
  <script src="http://ajax.aspnetcdn.com/ajax/jQuery/jquery-1.7.min.js"></script>
  <script>window.jQuery || document.write('<script src="libraries/widgets/quiz/js/jquery.js"><\/script>')</script> 
  <script data-main="libraries/frameworks/io2012/js/slides" 
    src="libraries/frameworks/io2012/js/require-1.0.8.min.js">
  </script>
  
  

</head>
<body style="opacity: 0">
  <slides class="layout-widescreen">
    
    <!-- LOGO SLIDE -->
        <slide class="title-slide segue nobackground">
  <hgroup class="auto-fadein">
    <h1>Interactive and exploratory visualization of epigenome-wide data</h1>
    <h2>Molecular Medicine Tri-Conference, February 18, 2015</h2>
    <p>Hector Corrada Bravo<br/>Center for Bioinformatics and Computational Biology, University of Maryland</p>
  </hgroup>
  <article></article>  
</slide>
    

    <!-- SLIDES -->
    <slide class="" id="slide-1" style="background:;">
  <hgroup>
    <h2>Our motivation</h2>
  </hgroup>
  <article data-timings="">
    <p>Measuring DNA methylation and understanding role in expression regulation in solid tumors</p>

<div class="centered">
<img src="images/block.png" style="max-height: 100%; max-width: 100%"/>
</div>

<p>Large blocks of <em>hypo-methylation</em> (sometimes Mbps long) in colon cancer</p>

<ul>
<li><strong>Hansen, et al., <em>Nat. Genetics</em>, 2011</strong></li>
<li>Corrada Bravo, et al., <em>BMC Bioinformatics</em>, 2012</li>
<li>Timp, et al., <em>Genome Medicine</em>, 2014</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-2" style="background:;">
  <hgroup>
    <h2>Our motivation</h2>
  </hgroup>
  <article data-timings="">
    <p>Measuring DNA methylation and understanding role in expression regulation in solid tumors</p>

<div class="centered">
<img src="images/antiprofile.png" style="max-height: 45%; max-width: 45%"/>
</div>

<p>Genes with <em>hyper-variable</em> expression in colon cancer are enriched within these blocks.</p>

<ul>
<li><strong>Hansen, et al., <em>Nat. Genetics</em>, 2011</strong></li>
<li>Corrada Bravo, et al., <em>BMC Bioinformatics</em>, 2012</li>
<li>Timp, et al., <em>Genome Medicine</em>, 2014</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-3" style="background:;">
  <hgroup>
    <h2>Our motivation</h2>
  </hgroup>
  <article data-timings="">
    <p>Measuring DNA methylation and understanding role in expression regulation in solid tumors</p>

<div class="centered">
<img src="images/barcode.png" style="max-height: 45%; max-width: 45%"/>
<img src="images/barcode2.png" style="max-height: 45%; max-width: 45%"/>
</div>

<p>Genes with <em>consistent hyper-variable expression</em> across tumors are tissue-specific.</p>

<ul>
<li>Hansen, et al., <em>Nat. Genetics</em>, 2011</li>
<li><strong>Corrada Bravo, et al., <em>BMC Bioinformatics</em>, 2012</strong></li>
<li>Timp, et al., <em>Genome Medicine</em>, 2014</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-4" style="background:;">
  <hgroup>
    <h2>Our motivation</h2>
  </hgroup>
  <article data-timings="">
    <p>Measuring DNA methylation and understanding role in expression regulation in solid tumors</p>

<div class="centered">
<img src="images/aucs.png" style="max-height: 30%; max-width: 30%"/>
</div>

<p>Consistent hyper-variability can be used as stable signatures.</p>

<ul>
<li>Hansen, et al., <em>Nat. Genetics</em>, 2011</li>
<li><strong>Corrada Bravo, et al., <em>BMC Bioinformatics</em>, 2012</strong></li>
<li>Timp, et al., <em>Genome Medicine</em>, 2014</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-5" style="background:;">
  <hgroup>
    <h2>Our motivation</h2>
  </hgroup>
  <article data-timings="">
    <p>Measuring DNA methylation and understanding role in expression regulation in solid tumors</p>

<div class="centered">
<img src="images/relapse.png" style="max-height: 30%; max-width: 30%"/>
</div>

<p>Degree of expression deviation from normality in specific genes can be used as prognostic tool.</p>

<ul>
<li>Hansen, et al., <em>Nat. Genetics</em>, 2011</li>
<li>Corrada Bravo, et al., <em>BMC Bioinformatics</em>, 2012</li>
<li><strong>Dinalankara and Corrada Bravo, <em>Cancer Informatics</em>, in press</strong></li>
<li>Timp, et al., <em>Genome Medicine</em>, 2014</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-6" style="background:;">
  <hgroup>
    <h2>Our motivation</h2>
  </hgroup>
  <article data-timings="">
    <p>Measuring DNA methylation and understanding role in expression regulation in solid tumors</p>

<div class="centered">
<img src="images/minfi.png" style="max-height: 60%; max-width: 60%"/>
</div>

<p>Blocks can be detected using Illumina HumanMethylation450k bead arrays.</p>

<ul>
<li>Hansen, et al., <em>Nat. Genetics</em>, 2011</li>
<li>Corrada Bravo, et al., <em>BMC Bioinformatics</em>, 2012</li>
<li><strong>Timp, et al., <em>Genome Medicine</em>, 2014</strong></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-7" style="background:;">
  <hgroup>
    <h2>Our motivation</h2>
  </hgroup>
  <article data-timings="">
    <p>Measuring DNA methylation and understanding role in expression regulation in solid tumors</p>

<div class="centered">
<img src="images/timp1.png" style="max-height: 70%; max-width: 70%"/>
</div>

<p>Hypo-methylation blocks occur in all tumor types. Some universal, many cancer-specific</p>

<ul>
<li>Hansen, et al., <em>Nat. Genetics</em>, 2011</li>
<li>Corrada Bravo, et al., <em>BMC Bioinformatics</em>, 2012</li>
<li><strong>Timp, et al., <em>Genome Medicine</em>, 2014</strong></li>
</ul>

<h2>Our motivation</h2>

<p>Measuring DNA methylation and understanding role in expression regulation in solid tumors</p>

<div class="centered">
<img src="images/timp2.png" style="max-height: 70%; max-width: 70%"/>
</div>

<p>Gene expression <em>hyper-variability</em> is enriched within <em>hypo-methylation</em> blocks</p>

<ul>
<li>Hansen, et al., <em>Nat. Genetics</em>, 2011</li>
<li>Corrada Bravo, et al., <em>BMC Bioinformatics</em>, 2012</li>
<li><strong>Timp, et al., <em>Genome Medicine</em>, 2014</strong></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-8" style="background:;">
  <hgroup>
    <h2>Our motivation</h2>
  </hgroup>
  <article data-timings="">
    <p>Measuring DNA methylation and understanding role in expression regulation in solid tumors</p>

<div class="centered">
<img src="images/timp3.png" style="max-height: 70%; max-width: 70%"/>
</div>

<p>Degree of hypo-methylation increases through cancer progression in blocks.</p>

<ul>
<li>Hansen, et al., <em>Nat. Genetics</em>, 2011</li>
<li>Corrada Bravo, et al., <em>BMC Bioinformatics</em>, 2012</li>
<li><strong>Timp, et al., <em>Genome Medicine</em>, 2014</strong></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-9" style="background:;">
  <hgroup>
    <h2>Summary</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Large regions of hypo-methylation seems to be consistent in cancer

<ul>
<li>occur in pre-cancerous lesions</li>
<li>hypo-methylation increases with cancer progression</li>
</ul></li>
<li>Gene expression hyper-variability enriched within these regions

<ul>
<li>tissue-specific genes enriched within these regions</li>
<li>can use degree of deviation from normality as stable diagnosis and prognosis mark in multiple cancer types</li>
</ul></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-10" style="background:;">
  <hgroup>
    <h2>R/Bioconductor</h2>
  </hgroup>
  <article data-timings="">
    <div class="centered">
<img src="images/rbioc.png" style="max-height: 70%; max-width: 70%"/>
</div>

<ul>
<li>State-of-the-art computational and statistical analysis
platform</li>
<li>We develop and apply methods for these analyses in this platform</li>
<li>Our collaborators take part in analysis in this platform</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-11" style="background:;">
  <hgroup>
    <h2>What we wanted</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>Data transformation and modeling: data smoothing, region finding (R/Bioconductor: <code>Bsmooth</code>, <code>minfi</code>)</li>
<li>Genome browsing: search by gene, search by overlap</li>
<li>Region analysis: overlap with other data (our own, other labs, UCSC, ensembl)</li>
<li>Regulation: expression data (Gene Expression Barcode)</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-12" style="background:;">
  <hgroup>
    <h2>Analysis era</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>Funding agencies calling for proposals to (strictly) <em>analyze</em> project data

<ul>
<li>Epigenomics roadmap, Encode, TCGA, ...</li>
</ul></li>
<li>Journals calling for (strictly) <em>analysis</em> papers (e.g., Nature Methods)</li>
<li><em>We have unprecedented ability to measure</em></li>
<li><em>and lots of publicly available data to contextualize it</em></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-13" style="background:;">
  <hgroup>
    <h2>Analysis era</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Funding agencies calling for proposals to (strictly) <em>analyze</em> project data</li>
<li>Epigenomics roadmap, Encode, TCGA, ...</li>
<li>Journals calling for (strictly) <em>analysis</em> papers (e.g., Nature Methods)</li>
<li><em>We have unprecedented ability to measure</em></li>
<li><em>and lots of publicly available data to contextualize it</em></li>
</ul>

<div class="centered">
<img src="images/hadley.png" style="max-height: 50%; max-width: 50%"/>
<img src="images/hadley2.png" style="max-height: 30%; max-width: 30%"/>
<footer class="source">[H. Wickham]</footer>
</div>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-14" style="background:;">
  <article data-timings="">
    <div class="centered">
<img src="images/epiviz_2_logo_medium.png" style="max-height: 40%; max-width: 40%"/>
</div>

<h3>Integrative, visual and computational exploratory analysis of genomic data</h3>

<ul>
<li>Browser-based</li>
<li>Interactive</li>
<li>Integration of data</li>
<li>Reproducible dissemination</li>
<li>Communication with R/Bioc: <code>epivizr</code> package</li>
</ul>

<blockquote>
<p><strong>I want to use a genome browser track as a display device in R!!</strong></p>
</blockquote>

<p>e.g.: <a href="http://epiviz.cbcb.umd.edu/2?ws=45KBV4C7z3u">http://epiviz.cbcb.umd.edu/2?ws=45KBV4C7z3u</a>
<footer class="source">[Nat. Methods, 2014]</footer></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-15" style="background:;">
  <article data-timings="">
    <iframe data-src="http://epiviz.cbcb.umd.edu/2/?ws=W3ieGz19icm" width="99%"></iframe>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-16" style="background:;">
  <hgroup>
    <h2>Communication with R/Bioc</h2>
  </hgroup>
  <article data-timings="">
    <p>Using the <code>epivizr</code> package</p>

<ul>
<li>Setup up an <code>epivizr</code> session</li>
</ul>

<pre><code class="r">mgr &lt;- startEpiviz(workspace=&quot;qyOTB6vVnff&quot;)
</code></pre>

<ul>
<li>Add a device with <code>GRanges</code> data</li>
</ul>

<pre><code class="r">blocks_dev &lt;- mgr$addDevice(colon_blocks, &quot;450k blocks&quot;)
</code></pre>

<ul>
<li>Subset ranges by width</li>
</ul>

<pre><code class="r">keep &lt;- width(colon_blocks) &gt; 250000
mgr$updateDevice(blocks_dev, colon_blocks[keep,])
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-17" style="background:;">
  <hgroup>
    <h2>Communication with R/Bioc</h2>
  </hgroup>
  <article data-timings="">
    <p>Using the <code>epivizr</code> package: browse by regions of interest.</p>

<ul>
<li>What&#39;s around the widest blocks?</li>
</ul>

<pre><code class="r">o &lt;- order(-width(colon_blocks))
slideShowRegions &lt;- colon_blocks[o[1:5],]
slideShowRegions &lt;- slideShowRegions + 1e5
mgr$slideshow(slideShowRegions)
</code></pre>

<blockquote>
<p><code>epivizr</code> uses WebSockets for connection, same as <code>shiny</code>. Big, big, big
thanks to the @rstudio folks for working on this infrastructure.</p>
</blockquote>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-18" style="background:;">
  <hgroup>
    <h2>Plugins, plugins, plugins</h2>
  </hgroup>
  <article data-timings="">
    <p>Our architecture is <em>dynamically extensible</em>. We can easily integrate new data types and add new visualizations.</p>

<p>Example: adding a new visualization</p>

<p>see: <a href="https://gist.github.com/11017650">https://gist.github.com/11017650</a></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-19" style="background:;">
  <article data-timings="">
    <p><a href="http://epiviz.cbcb.umd.edu/?gist%5B%5D=11017650&amp;ws=Y8kWxCO2Ajn">http://epiviz.cbcb.umd.edu/?gist[]=11017650&amp;ws=Y8kWxCO2Ajn</a></p>

<iframe data-src="http://epiviz.cbcb.umd.edu/2/?gist[]=11017650&ws=Y8kWxCO2Ajn"></iframe>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-20" style="background:;">
  <hgroup>
    <h2>Plugins, plugins, plugins</h2>
  </hgroup>
  <article data-timings="">
    <p>Example: import data from UCSC browser</p>

<p>see: <a href="https://gist.github.com/c41a2df3671395d8e4ad">https://gist.github.com/c41a2df3671395d8e4ad</a></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-21" style="background:;">
  <article data-timings="">
    <p><a href="http://epiviz.cbcb.umd.edu/?ws=cX1PgToUQs&amp;seqName=chr11&amp;start=59463945&amp;end=60638081&amp;gist%5B%5D=c41a2df3671395d8e4ad&amp;settings=default&amp;">http://epiviz.cbcb.umd.edu/?ws=cX1PgToUQs&amp;seqName=chr11&amp;start=59463945&amp;end=60638081&amp;gist[]=c41a2df3671395d8e4ad&amp;settings=default&amp;</a></p>

<iframe data-src="http://epiviz.cbcb.umd.edu/2/?ws=cX1PgToUQs&seqName=chr11&start=59463945&end=60638081&gist[]=c41a2df3671395d8e4ad&settings=default&"></iframe>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-22" style="background:;">
  <hgroup>
    <h2>Plugins, plugins, plugins</h2>
  </hgroup>
  <article data-timings="">
    <iframe data-src="http://epiviz.github.io"></iframe>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-23" style="background:;">
  <hgroup>
    <h1>Datatypes</h1>
  </hgroup>
  <article data-timings="">
    <div class="centered">
<img src="images/datatypes.png" style="max-width=50%; min-width=50%"/>
</div>

<ul>
<li>Based on &quot;three-table&quot; design</li>
<li>Scripts can define coordinate space</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-24" style="background:;">
  <article data-timings="">
    <p><a href="http://epiviz.cbcb.umd.edu/?ws=SRHZlWRRAPd&amp;gist%5B%5D=a82a998817564ce3fe48&amp;settings=default&amp;">http://epiviz.cbcb.umd.edu/?ws=SRHZlWRRAPd&amp;gist[]=a82a998817564ce3fe48&amp;settings=default&amp;</a></p>

<iframe data-src="http://epiviz.cbcb.umd.edu/2/?ws=SRHZlWRRAPd&gist[]=a82a998817564ce3fe48&settings=default&"></iframe>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-25" style="background:;">
  <hgroup>
    <h2>Build your own browser</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Standalone version (no internet required, javascript code provided in <code>epivizr</code>)</li>
<li>Browse your favorite genome:</li>
</ul>

<pre><code class="r">library(epivizr)
library(Mus.musculus)

mgr &lt;- startStandalone(geneInfo=Mus.musculus, geneInfoName=&quot;mm10&quot;,
                          keepSeqlevels=paste0(&quot;chr&quot;,c(1:19,&quot;X&quot;,&quot;Y&quot;)))
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-26" style="background:;">
  <hgroup>
    <h2>Analysis era</h2>
  </hgroup>
  <article data-timings="">
    <div class="centered">
<img src="images/hadley.png" style="max-height: 50%; max-width: 50%"/>
<img src="images/hadley2.png" style="max-height: 30%; max-width: 30%"/>
<footer class="source">[H. Wickham]</footer>
</div>

<p>One interpretation of <em>Big Data</em> is <em>Many relevant sources of contextual data</em></p>

<ul>
<li>Easily access/integrate <em>contextual</em> data</li>
<li>Driven by exploratory analysis of <em>immediate</em> data</li>
<li>Iterative process</li>
<li>Visual and computational exploration go hand in hand</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-27" style="background:;">
  <hgroup>
    <h2>Creativity in exploration</h2>
  </hgroup>
  <article data-timings="">
    <p>We are building a software system to support creative exploratory analysis of epigenome-wide datasets...</p>

<p><div class="centered">
<img src="images/terry.png" style="max-height: 60%; max-width: 60%"/>
<footer class="source">[T. Speed]</footer></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-28" style="background:;">
  <hgroup>
    <h2>Visualization goals</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Context 

<ul>
<li>Integrate and align multiple data sources; navigate; search</li>
<li><em>Connect</em>: brushing</li>
<li><em>Encode</em>: map visualization properties to data on the fly</li>
<li><em>Reconfigure</em>: multiple views of the same data</li>
</ul></li>
</ul>

<p><footer class="source">[Perer &amp; Shneiderman]</footer></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-29" style="background:;">
  <hgroup>
    <h2>Visualization goals</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Data

<ul>
<li><em>Select and filter</em>: tight-knit integration with R/Bioconductor; </li>
<li>(future) filters on visualization propagate to data environment</li>
</ul></li>
<li>Model

<ul>
<li>New &#39;measurements&#39; the result of modeling; perhaps suggested by data context</li>
</ul></li>
</ul>

<p><footer class="source">[Perer &amp; Shneiderman]</footer></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-30" style="background:;">
  <hgroup>
    <h2>Check it out:</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li><a href="http://epiviz.github.io">http://epiviz.github.io</a></li>
<li><a href="http://epiviz.cbcb.umd.edu">http://epiviz.cbcb.umd.edu</a></li>
<li><a href="http://github.com/epiviz">http://github.com/epiviz</a></li>
</ul>

<p>Nature Methods 2014<br>
Follow us: @epiviz  </p>

<p>These slides available: <a href="http://hcorrada.github.io/tricon2015">http://hcorrada.github.io/tricon2015</a></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-31" style="background:;">
  <hgroup>
    <h2>Acknowledgements</h2>
  </hgroup>
  <article data-timings="">
    <p><img src="http://www.cs.umd.edu/~florinc/img/florin_chelaru.jpg"/>
Florin Chelaru, UMD</p>

<ul>
<li>CBCB@UMD</li>
<li>JHU/Harvard: Kasper Hansen, Winston Timp, Rafael Irizarry, Andy Feinberg</li>
<li>Genentech: Michael Lawrence</li>
<li>Rstudio: Joe Cheng, et al.</li>
<li>Funding: NIH, Genentech</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

    <slide class="backdrop"></slide>
  </slides>
  <div class="pagination pagination-small" id='io2012-ptoc' style="display:none;">
    <ul>
      <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=1 title='Our motivation'>
         1
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=2 title='Our motivation'>
         2
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=3 title='Our motivation'>
         3
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=4 title='Our motivation'>
         4
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=5 title='Our motivation'>
         5
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=6 title='Our motivation'>
         6
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=7 title='Our motivation'>
         7
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=8 title='Our motivation'>
         8
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=9 title='Summary'>
         9
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=10 title='R/Bioconductor'>
         10
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=11 title='What we wanted'>
         11
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=12 title='Analysis era'>
         12
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=13 title='Analysis era'>
         13
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=14 title=''>
         14
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=15 title=''>
         15
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=16 title='Communication with R/Bioc'>
         16
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=17 title='Communication with R/Bioc'>
         17
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=18 title='Plugins, plugins, plugins'>
         18
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=19 title=''>
         19
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=20 title='Plugins, plugins, plugins'>
         20
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=21 title=''>
         21
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=22 title='Plugins, plugins, plugins'>
         22
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=23 title='Datatypes'>
         23
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=24 title=''>
         24
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=25 title='Build your own browser'>
         25
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=26 title='Analysis era'>
         26
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=27 title='Creativity in exploration'>
         27
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=28 title='Visualization goals'>
         28
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=29 title='Visualization goals'>
         29
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=30 title='Check it out:'>
         30
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=31 title='Acknowledgements'>
         31
      </a>
    </li>
  </ul>
  </div>  <!--[if IE]>
    <script 
      src="http://ajax.googleapis.com/ajax/libs/chrome-frame/1/CFInstall.min.js">  
    </script>
    <script>CFInstall.check({mode: 'overlay'});</script>
  <![endif]-->
</body>
  <!-- Load Javascripts for Widgets -->
  <script src="libraries/widgets/bootstrap/js/bootstrap.min.js"></script>
<script src="libraries/widgets/bootstrap/js/bootbox.min.js"></script>

  <script>  
  $(function (){ 
    $("#example").popover(); 
    $("[rel='tooltip']").tooltip(); 
  });  
  </script>  
  <!-- LOAD HIGHLIGHTER JS FILES -->
  <script src="libraries/highlighters/highlight.js/highlight.pack.js"></script>
  <script>hljs.initHighlightingOnLoad();</script>
  <!-- DONE LOADING HIGHLIGHTER JS FILES -->
   
  </html>